File: OOP based implmentation/OOP based implmentation/Project/objs/Student.java
package objs;
import java.io.BufferedWriter;
import java.io.FileWriter;
import java.io.IOException;
import login.StudentSignup;

public class Student {
    private StudentSignup signupDetails;
    private StudentPersonal personalDetails;
    private StudentAcademics academicDetails;

    public Student(StudentSignup signupDetails, StudentPersonal personalDetails, StudentAcademics academicDetails) {
        this.signupDetails = signupDetails;
        this.personalDetails = personalDetails;
        this.academicDetails = academicDetails;
    }

    public void saveAllDetailsToFile(String filename) {
        try (BufferedWriter writer = new BufferedWriter(new FileWriter(filename, true))) {
            writer.write("Username: " + signupDetails.getUsername() + ", ");
            writer.write("Password: " + signupDetails.getPassword() + ", ");
            writer.write("Roll Number: " + signupDetails.getRollNum() + ", ");
            writer.write("Course: " + signupDetails.getCourse() + "\n");
            
            writer.write("Name: " + personalDetails.getName() + ", ");
            writer.write("Age: " + personalDetails.getAge() + ", ");
            writer.write("Gender: " + personalDetails.getGender() + ", ");
            writer.write("Address: " + personalDetails.getAddress() + ", ");
            writer.write("Hobby: " + personalDetails.getHobby() + "\n");

            writer.write("CGPA: " + academicDetails.getCgpa() + ", ");
            writer.write("Skills: " + academicDetails.getSkills() + ", ");
            writer.write("Test Score: " + academicDetails.getTestScore() + ", ");
            writer.write("Expected CTC: " + academicDetails.getExpectedCTC() + ", ");
      
            writer.write("--------------------------------------------------\n");
            System.out.println("All student details saved successfully.");
        } catch (IOException e) {
            System.out.println("An error occurred while saving student details.");
            e.printStackTrace();
        }
    }
}

File: OOP based implmentation/OOP based implmentation/Project/objs/Student.cpp
#include <iostream>
#include <fstream>
#include <string>
#include "Student.h"
#include <limits>

using namespace std;

Student::Student(StudentSignup &signupDetails, StudentPersonal &personalDetails, StudentAcademics &academicDetails)
    : signupDetails(signupDetails), personalDetails(personalDetails), academicDetails(academicDetails)
{
}

void Student::saveAllDetailsToFile(const string &filename)
{
    ofstream writer(filename, ios_base::app);
    if (!writer.is_open())
    {
        cout << "An error occurred while saving student details." << endl;
        return;
    }

    writer << "Username: " << signupDetails.getUsername() << ", ";
    writer << "Password: " << signupDetails.getPassword() << ", ";
    writer << "Roll Number: " << signupDetails.getRollNum() << ", ";
    writer << "Course: " << signupDetails.getCourse() << "\n";

    writer << "Name: " << personalDetails.getName() << ", ";
    writer << "Age: " << personalDetails.getAge() << ", ";
    writer << "Gender: " << personalDetails.getGender() << ", ";
    writer << "Address: " << personalDetails.getAddress() << ", ";
    writer << "Hobby: " << personalDetails.getHobby() << "\n";

    writer << "CGPA: " << academicDetails.getCgpa() << ", ";
    writer << "Skills: " << academicDetails.getSkills() << ", ";
    writer << "Test Score: " << academicDetails.getTestScore() << ", ";
    writer << "Expected CTC: " << academicDetails.getExpectedCTC() << ", ";

    writer << "--------------------------------------------------\n";
    cout << "All student details saved successfully." << endl;
    writer.close();
}